--- postgresql/src/backend/storage/lmgr/lwlock.c
+++ postgresql-wasm/src/backend/storage/lmgr/lwlock.c
@@ -129,7 +129,11 @@
  * All these names are user-visible as wait event names, so choose with care
  * ... and do not forget to update the documentation's list of wait events.
  */
+#if !defined(__wasi__)
 extern const char *const IndividualLWLockNames[];	/* in lwlocknames.c */
+#else
+#include "lwlocknames.c"
+#endif
 
 static const char *const BuiltinTrancheNames[] = {
 	/* LWTRANCHE_XACT_BUFFER: */
