--- duckdb-wasm/submodules/duckdb/src/include/duckdb/common/file_open_flags.hpp	2024-06-05 23:12:32.071226596 +0200
+++ duckdb-pglite/submodules/duckdb/src/include/duckdb/common/file_open_flags.hpp	2024-06-05 09:22:15.647820784 +0200
@@ -27,6 +27,8 @@
 	static constexpr idx_t FILE_FLAGS_NULL_IF_NOT_EXISTS = idx_t(1 << 7);
 	static constexpr idx_t FILE_FLAGS_PARALLEL_ACCESS = idx_t(1 << 8);
 
+	idx_t flags = 0;
+
 public:
 	FileOpenFlags() = default;
 	constexpr FileOpenFlags(idx_t flags) : flags(flags) { // NOLINT: allow implicit conversion
@@ -101,7 +103,7 @@
 	}
 
 private:
-	idx_t flags = 0;
+
 	FileLockType lock = FileLockType::NO_LOCK;
 	FileCompressionType compression = FileCompressionType::UNCOMPRESSED;
 };
