--- postgresql-16.2/src/backend/snowball/dict_snowball.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/backend/snowball/dict_snowball.c	2024-04-29 16:44:45.641372431 +0200
@@ -10,6 +10,7 @@
  *
  *-------------------------------------------------------------------------
  */
+
 #include "postgres.h"
 
 #include "commands/defrem.h"
@@ -24,6 +25,8 @@
 #undef MININT
 #endif
 
+#include PG_PLUGIN_INCLUDE
+
 /* Now we can include the original Snowball header.h */
 #include "snowball/libstemmer/header.h"
 #include "snowball/libstemmer/stem_ISO_8859_1_basque.h"
@@ -158,6 +161,11 @@
 	{NULL, 0, NULL, NULL, NULL} /* list end marker */
 };
 
+static char *
+nofnptr_lowerstr(const char *str)
+{
+	return lowerstr_with_len(str, strlen(str));
+}
 
 typedef struct DictSnowball
 {
@@ -216,6 +224,7 @@
 					lang, GetDatabaseEncodingName())));
 }
 
+
 Datum
 dsnowball_init(PG_FUNCTION_ARGS)
 {
@@ -236,7 +245,7 @@
 				ereport(ERROR,
 						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 						 errmsg("multiple StopWords parameters")));
-			readstoplist(defGetString(defel), &d->stoplist, lowerstr);
+			readstoplist(defGetString(defel), &d->stoplist, nofnptr_lowerstr);
 			stoploaded = true;
 		}
 		else if (strcmp(defel->defname, "language") == 0)
@@ -261,7 +270,7 @@
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 				 errmsg("missing Language parameter")));
 
-	d->dictCtx = CurrentMemoryContext;
+	d->dictCtx = getCurrentMemoryContext();
 
 	PG_RETURN_POINTER(d);
 }
@@ -315,10 +324,10 @@
 		}
 
 		/* see comment about d->dictCtx */
-		saveCtx = MemoryContextSwitchTo(d->dictCtx);
+		saveCtx = setMemoryContextSwitch(d->dictCtx);
 		SN_set_current(d->z, strlen(txt), (symbol *) txt);
 		d->stem(d->z);
-		MemoryContextSwitchTo(saveCtx);
+		setMemoryContextSwitch(saveCtx);
 
 		if (d->z->p && d->z->l)
 		{
