--- postgresql-16.2/src/include/executor/executor.h	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/include/executor/executor.h	2024-04-29 16:40:54.899655332 +0200
@@ -20,7 +20,7 @@
 #include "nodes/parsenodes.h"
 #include "utils/memutils.h"
 
-
+#include PG_PLUGIN_INCLUDE
 /*
  * The "eflags" argument to ExecutorStart and the various ExecInitNode
  * routines is a bitwise OR of the following flag bits, which tell the
@@ -351,9 +351,9 @@
 	Datum		retDatum;
 	MemoryContext oldContext;
 
-	oldContext = MemoryContextSwitchTo(econtext->ecxt_per_tuple_memory);
+	oldContext = setMemoryContextSwitch(econtext->ecxt_per_tuple_memory);
 	retDatum = state->evalfunc(state, econtext, isNull);
-	MemoryContextSwitchTo(oldContext);
+	setCurrentMemoryContext(oldContext);
 	return retDatum;
 }
 #endif
