--- postgresql-16.2/src/backend/postmaster/postmaster.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/backend/postmaster/postmaster.c	2024-05-06 10:08:13.589152956 +0200
@@ -1533,7 +1533,7 @@
 	/* Locate the postgres executable itself */
 	if (find_my_exec(argv0, my_exec_path) < 0)
 		ereport(FATAL,
-				(errmsg("%s: could not locate my own executable path", argv0)));
+				(errmsg("%s:1536: could not locate my own executable path", argv0)));
 
 #ifdef EXEC_BACKEND
 	/* Locate executable backend before we change working directory */
