--- postgresql-16.2/src/common/logging.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/common/logging.c	2024-05-06 13:23:40.756348647 +0200
@@ -79,7 +79,7 @@
 /*
  * This should be called before any output happens.
  */
-void
+EMSCRIPTEN_KEEPALIVE void
 pg_logging_init(const char *argv0)
 {
 	const char *pg_color_env = getenv("PG_COLOR");
@@ -201,7 +201,7 @@
 	log_locus_callback = cb;
 }
 
-void
+EMSCRIPTEN_KEEPALIVE void
 pg_log_generic(enum pg_log_level level, enum pg_log_part part,
 			   const char *pg_restrict fmt,...)
 {
