--- duckdb-wasm/submodules/duckdb/src/core_functions/aggregate/distributive/arg_min_max.cpp	2024-06-05 23:12:32.042226989 +0200
+++ duckdb-pglite/submodules/duckdb/src/core_functions/aggregate/distributive/arg_min_max.cpp	2024-06-05 08:25:57.963388434 +0200
@@ -164,7 +164,7 @@
 		if (!state.is_initialized || state.arg_null) {
 			finalize_data.ReturnNull();
 		} else {
-			STATE::template ReadValue(finalize_data.result, state.arg, target);
+			STATE::template ReadValue<>(finalize_data.result, state.arg, target);
 		}
 	}
 
@@ -248,7 +248,7 @@
 			return;
 		}
 		if (!target.is_initialized || COMPARATOR::Operation(source.value, target.value)) {
-			STATE::template AssignValue(target.value, source.value);
+			STATE::template AssignValue<>(target.value, source.value);
 			AssignVector(target, *source.arg, source.arg_null, 0);
 			target.is_initialized = true;
 		}
