diff --git a/src/backend/Makefile b/src/backend/Makefile
index 82cae98a44..5be25c17ac 100644
--- a/src/backend/Makefile
+++ b/src/backend/Makefile
@@ -60,6 +60,7 @@ override LDFLAGS := $(LDFLAGS) $(LDFLAGS_EX) $(LDFLAGS_EX_BE)
 
 all: submake-libpgport submake-catalog-headers submake-utils-headers postgres $(POSTGRES_IMP)
 
+ifneq ($(PORTNAME), emscripten)
 ifneq ($(PORTNAME), cygwin)
 ifneq ($(PORTNAME), win32)
 ifneq ($(PORTNAME), aix)
@@ -70,7 +71,20 @@ postgres: $(OBJS)
 endif
 endif
 endif
-
+endif
+ifeq ($(PORTNAME), emscripten)
+PGLITE += $(top_builddir)/libpglite.a
+PGLITE += $(top_builddir)/src/common/libpgcommon_srv.a
+PGLITE += $(top_builddir)/src/port/libpgport_srv.a
+PGMAIN = main/main.o tcop/postgres.o
+postgres: $(OBJS)
+	$(warning "======================== libpglite.a =====================")
+	#	emar cr $(filter-out $(PGMAIN),$(filter %.o,$(call expand_subsys,$^)))
+	@python3 $(top_builddir)/../../wasm-build/link-pglite.py $(call expand_subsys,$^)
+	$(warning "======================= linking postgres main =====================")
+	COPTS="-Oz -g0 --closure 1" $(CC) $(MAIN_MODULE) $(CFLAGS) $(LDFLAGS) -o $@ $(PGMAIN) $(PGLITE) $(LIBS)
+	$(warning "========================= done ============================")
+endif
 ifeq ($(PORTNAME), cygwin)
 
 postgres: $(OBJS)
